src/main/java/org/elasticsearch/index/IndexService.java

```diff
@@ -388,7 +388,7 @@ private void closeShardInjector(String reason, ShardId sId, Injector shardInject
             // and close the shard so no operations are allowed to it
             if (indexShard != null) {
                 try {
-                    indexShard.close(reason);
+                    indexShard.close(reason, deleted.get());
```

Comment:
the deleted flag is only set when an index is deleted. The PR description says we don't want to flush upon relocation. I'm a bit on the fence w.r.t flush on relocation. On one hand it would be good to leave things consistent. On the other hand it may delay things needlessly (this runs on the cluster state thread). In anyway we should change the description of the PR we keep this way..